File: backend/src/entities/AuditLog.ts
import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';

@Entity('audit_log')
export class AuditLog {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ type: 'enum', enum: ['manager', 'auto', 'student'] })
  actor: 'manager' | 'auto' | 'student';

  @Column()
  action: string;

  @Column()
  entity: string;

  @Column()
  entity_id: number;

  @Column({ type: 'json', nullable: true })
  before: any;

  @Column({ type: 'json', nullable: true })
  after: any;

  @Column({ type: 'timestamp', default: () => 'CURRENT_TIMESTAMP' })
  timestamp: Date;
}

File: backend/app/models/audit_log.py
from sqlalchemy import Column, Integer, String, DateTime, JSON
from .base import Base
from datetime import datetime

class AuditLog(Base):
    __tablename__ = "audit_log"
    
    id = Column(Integer, primary_key=True, index=True)
    actor = Column(String, nullable=False)  # manager/auto/student
    action = Column(String, nullable=False)
    entity = Column(String, nullable=False)
    entity_id = Column(Integer, nullable=False)
    before = Column(JSON, nullable=True)
    after = Column(JSON, nullable=True)
    timestamp = Column(DateTime, nullable=False, default=datetime.utcnow, index=True)